data/transactions/logic/eval.go

```diff
@@ -224,7 +232,7 @@ func Eval(program []byte, params EvalParams) (pass bool, err error) {
 		if cx.stepCount > len(cx.program) {
 			return false, errLoopDetected
 		}
-		if params.Proto != nil && uint64(cx.cost) > params.Proto.LogicSigMaxCost {
+		if params.Proto == nil || uint64(cx.cost) > params.Proto.LogicSigMaxCost {
```

Comment:
maybe make one `if params.Proto == nil { return errProgrammingError }` check at the top and then the rest of the function can assume that params.Proto is not nil and not check it each time

Reply:
addressed.